---
layout: container
title: GSoC 2014
---

Last year we participated for the first time in the GSoC.
You can enjoy this visual presentation from our students about their projects.

<iframe width="560" height="315" src="https://www.youtube.com/embed/SkMKuj-Q0YI" frameborder="0" allowfullscreen style="margin: 0 auto; display: block"></iframe>  

Multiple sequence alignment viewer
----------------------------------

Multiple (i.e. >2) sequence (i.e. protein or nucleic acid) alignments (MSA) are used to investigate the evolutionary relationships between those sequences. Researchers use it to explore numerous questions such as: identifying conserved active sites and structural domains; training HMMs (Hidden Markov Models) to search for other members of the gene family when new sequencing is carried out; locating conserved lengths that may indicate important spatial structural roles; and building phylogenetic trees. The alignment step itself is algorithmic, however the automatically generated output needs to be checked manually because none of the available computer programs are as good as an experienced scientist. Thus the need for visualization.

* [Blog](https://sebi.wilzbach.me/blog/articles/categories/biojs/index.html)  
* [Component](http://biojs.io/d/msa)

<iframe src="http://workmen.biojs.net/demo/msa/fer1" width="100%" height="400px"></iframe>
					
Phylogenetic Tree
-----------------

In bioinformatics and computational biology, the evolution of genes and species is often represented using dendrograms called phylogenetic trees. As the history of gene families can be complex (including lots of gene duplications or losses in different moments of its history) its visualisation can become a difficult task. A good/accurate visualisation of phylogenetic trees allows easier understanding and interpretation of these trees helping to reveal the mechanisms that shape the evolution of a specific set of gene/species. Current web technologies represent a great framework for this kind of representations and we envisage a great opportunity for someone interested in coding a visually appealing and useful javascript plug-in to represent trees on the web. As a primer, we have started a BioJS component, called treeWidget, to visualise phylogenetic trees on the web. Through its modularity, treeWidget can be easily customized to add sequence information to the displayed tree, e.g. protein domains and alignment conservation patterns. In this project we propose to extend the idea behind treeWidget adding a range of functionalities to improve the usability of the plug-in across different kind of tree visualizations.


* [Blog](http://gsoc.daviddao.de)   
* [Component](http://biojs.io/d/exelixis)

<iframe src="http://workmen.biojs.net/demo/exelixis/createAndUpdate" width="100%" height="200px"></iframe>

Proteome visualization
---------------------

Proteins are often called the building blocks within our bodies because they carry out many important functions such as growth and repair, cell signaling, catalyzing chemical reactions and so on. Some species been thoroughly researched over the years but some others are under-represented, while equally important to the wider scientific. For example, plants have been under studied but could be instrumental in the future for issues like food security. To help researchers focus on new target species, we want to visualize the existing taxonomic diversity in the worldâ€™s largest protein information resource, UniProt.

* [Component](http://biojs.io/d/biojs-vis-proteome)  

<iframe src="http://workmen.biojs.net/demo/biojs-vis-proteome/simple" width="100%" height="455px"></iframe>

Genetic variation viewer
------------------------

Have you ever wonder what make humans unique? The key is in our genetic variation. The study of human genetic variations help us to look backwards and forwards at human history. On one side, it helps to better understand the human evolution; for instance human migrations from one place to other as well as how different groups relate to each other. On the other hand, genetic variation is key to identify disease-associated genes as well as drug targets; furthermore, understanding variations could help to accurately diagnose a patient and define the procedure to follow, and from one patient we could move to a population affected by a particular disease.

* [Blog](http://genetic-variation-viewer-biojs.blogspot.com)  
* [Component:](http://biojs.io/d/biojs-vis-hgv)

<iframe src="http://workmen.biojs.net/demo/biojs-vis-hgv/simple_example" width="100%" height="400px"></iframe>

BAM
---

A BAM file (.bam) is a binary version of a SAM file that contains sequence alignment data. BAM is the key type of format for dealing with Next Generation Sequencing (NGS) data. Visualization of data in BAM files is crucial for interpretation of read alignments, variants, gene expression, etc. Some web tools have been developed to visualise BAM data on the web such as GBrowse, JBrowse, TGAC Browser etc. The problem with these visualisation solutions is that they are not necessarily easily extendable or documented. The BioJS visualisation framework builds on a common minimum standard for development of JavaScript functionality that makes it easy to share, reuse and create web components for visualisation of biological data. We propose to develop a BioJS component that can cope with the visualisation of BAM files.

* [Blog](http://biojs-samviewer.blogspot.co.uk)
* [Component](http://biojs.io/d/biojs-vis-sam)

<iframe src="http://workmen.biojs.net/demo/biojs-vis-sam/single" width="100%" height="400px"></iframe>
